### ISSUE TITLE
terraform_unused_declarations warning if I declare data that is only used in a different file

### ISSUE THREAD

**Issue by sblask** [terraform-linters/tflint#1159]
In `main.tf`:

```hcl
data "aws_iam_policy_document" "bucket" {

  statement {
    sid = "ObjectPermissions"
    principals {
      identifiers = [var.account_with_access]
      type        = "AWS"
    }
    actions = [
      "s3:DeleteObject",
      "s3:GetObject",
      "s3:GetObjectAcl",
      "s3:GetObjectVersionAcl",
      "s3:PutObject",
      "s3:PutObjectAcl",
      "S3:PutObjectVersionAcl",
      "s3:PutObjectTagging",
    ]
    resources = [
      "arn:aws:s3:::${var.bucket_name}/*"
    ]
  }

  statement {
    sid = "BucketPermissions"
    principals {
      identifiers = [var.account_with_access]
      type        = "AWS"
    }
    actions = [
      "s3:ListBucket",
    ]
    resources = [
      "arn:aws:s3:::${var.bucket_name}"
    ]
  }

}
```

In `outputs.tf`:

```hcl
output "document" {
  value = data.aws_iam_policy_document.bucket.json
}
```

I am getting:

```console
Warning: data "aws_iam_policy_document" "bucket" is declared but not used (terraform_unused_declarations)

  on tf/modules/s3_bucket/main.tf line 92:
  92: data "aws_iam_policy_document" "bucket" {

Reference: https://github.com/terraform-linters/tflint/blob/v0.30.0/docs/rules/terraform_unused_declarations.md
```

If I move the content of outputs.tf into main.tf the warning goes away, but It should work with keeping them separate?

## Version

```console
$ tflint -v
TFLint version 0.30.0

$ terraform -v
Terraform v1.0.2
on darwin_amd64
```

**Comment by sblask** [comment]
Ah, never mind, for some reason vim showed me the warning in the wrong file.
